src/python/pants/backend/docker/parser.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+import os.path
+import re
+from dataclasses import InitVar, dataclass, field
+from typing import Generator
+
+from dockerfile import Command, parse_string
+
+from pants.backend.docker.target_types import DockerImageSources
+from pants.engine.fs import DigestContents, GlobMatchErrorBehavior, PathGlobs
+from pants.engine.rules import Get, collect_rules, rule
+
+_pex_target_regexp = re.compile(
```

Comment:
It might be worth compiling this on first use, inside the rule. The `re` module maintains an internal cache of recently used compiled regexes, so performance is probably fine (and even if not, the cost of recompiling the regex is probably negligible compared to the rest of this).

The reason is: In pants v1 we noticed that a substantial amount of time on every pantsd restart was taken up by compiling module-level regexes on import.